test/test_roscpp/test/src/timer_callbacks.cpp

```diff
@@ -74,15 +74,13 @@ class SteadyTimerHelper
     void callback(const SteadyTimerEvent& e)
     {
       bool first = last_call_.isZero();
-      SteadyTime last_call = last_call_;
-      last_call_ = SteadyTime::now();
-      SteadyTime start = last_call_;
+      last_call_ = e.current_real;
 
       if (!first)
       {
-        if (fabsf(expected_next_call_.toSec() - start.toSec()) > 0.1f)
+        if (fabsf(e.current_expected.toSec() - e.current_real.toSec()) > 0.1f)
```

Comment:
I still like splitting up too early, too late into separate checks. Even if you don't relax the too-late threshold. It should never be called too early, so the check can be really tight.